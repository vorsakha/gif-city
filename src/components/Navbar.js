import React from 'react';
import { NavLink } from 'react-router-dom';

export default function Navbar() {
  return (
    <div>
      <h1
        className="mt-4 mb-5 col-md-12 text-info text-center"
        style={{ fontFamily: 'Permanent Marker', fontSize: '5em' }}>
        Gif City
      </h1>
      <nav className="d-flex justify-content-center">
        <ul className="d-flex flex-row justify-content-center ">
          <li>
            <NavLink className="btn btn-info nav-btn" style={{ boxShadow: 'none' }} exact to="/">
              Trending
            </NavLink>
          </li>
          <li>
            <NavLink
              className="btn btn-info nav-btn"
              style={{ boxShadow: 'none' }}
              exact
              to="/random">
              Random
            </NavLink>
          </li>
          <li>
            <NavLink
              className="btn btn-info nav-btn"
              style={{ boxShadow: 'none' }}
              exact
              to="/search">
              Search
            </NavLink>
          </li>
        </ul>
      </nav>
    </div>
  );
}
