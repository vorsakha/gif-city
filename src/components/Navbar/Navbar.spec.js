import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from './Navbar';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Navbar', () => {
  it('renders properly', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders menu items', () => {
    render(
      <Router>
        <Navbar />
      </Router>
    );

    expect(screen.getByText('Trending')).toBeInTheDocument();
    expect(screen.getByText('Random')).toBeInTheDocument();
    expect(screen.getByText('Search')).toBeInTheDocument();
  });
});
