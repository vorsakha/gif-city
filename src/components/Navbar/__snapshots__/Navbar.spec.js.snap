// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders properly 1`] = `
<div>
  <h1
    class="mt-4 mb-5 col-md-12 text-info text-center"
    style="font-family: Permanent Marker; font-size: 5em;"
  >
    Gif City
  </h1>
  <nav
    class="d-flex justify-content-center"
  >
    <ul
      class="d-flex flex-row justify-content-center "
    >
      <li>
        <a
          aria-current="page"
          class="btn btn-info nav-btn active"
          href="/"
          style="box-shadow: none;"
        >
          Trending
        </a>
      </li>
      <li>
        <a
          class="btn btn-info nav-btn"
          href="/random"
          style="box-shadow: none;"
        >
          Random
        </a>
      </li>
      <li>
        <a
          class="btn btn-info nav-btn"
          href="/search"
          style="box-shadow: none;"
        >
          Search
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
